### PR TITLE
Shifts Model

### PR DESCRIPTION
**Pull request by nawilliams**
Addresses #4 

Migration for shifts that have:
start_time
end_time

Validation to avoid impossible times.

Added paper_trail

### CODE REVIEW

File: db/migrate/20180214191046_create_shifts.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class CreateShifts < ActiveRecord::Migration[5.1]
+  def change
+    create_table :shifts do |t|
+      t.datetime :start_time
+      t.datetime :end_time
```
Comment (Anbranin) [inline db/migrate/20180214191046_create_shifts.rb:5]: I don't see a user_id in here -- if shifts do belong to users, you may want to add one.

**Review by Anbranin — CHANGES_REQUESTED** [review]

File: app/models/shift.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 class Shift < ApplicationRecord
   has_paper_trail
-  belongs_to :user
+  belongs_to :user, foreign_key: :id, class_name: 'User'
```
Comment (Anbranin) [inline app/models/shift.rb]: The foreign_key is _never_ going to be `:id`, unless you're doing something super wonky and uncool. You also do not have to specify the class name here because it is the default.
Let's break this down:
`belongs_to :user` means there is a column in the `shifts` table that is the `id` of a user.
`foreign_key: :id` specifies what this column is, and here you're telling rails that it's the `id` field. You're saying that the `id` field, then, is the `id` of a user. That is incorrect--every table in a database needs a unique identifying index, and that is, by default, the ID field--and there is no reason to change that in this application.
`class_name: 'User'` does no harm in this case but by saying `belongs_to :user` rails automatically assumes that the class name is `User`. You only have to specify the class name if the foreign key is something other than `user_id`.
tl;dr: put this back the way it was and add a `user_id` to the shifts table.

**Review by Anbranin — CHANGES_REQUESTED** [review]

**Review by Anbranin — APPROVED** [review]
